modules/framework/galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/internal/rbac/CacheRBACImpl.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.internal.rbac;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import dev.galasa.framework.spi.auth.AuthStoreException;
+import dev.galasa.framework.spi.auth.IAuthStoreService;
+import dev.galasa.framework.spi.auth.IUser;
+import dev.galasa.framework.spi.rbac.CacheRBAC;
+import dev.galasa.framework.spi.rbac.RBACException;
+import dev.galasa.framework.spi.rbac.RBACService;
+import dev.galasa.framework.spi.rbac.Role;
+
+public class CacheRBACImpl implements CacheRBAC {
+
+    private Map<String, List<String>> usersToActionsMap;
+    private IAuthStoreService authStoreService;
+    private RBACService rbacService;
+    
+    public CacheRBACImpl(IAuthStoreService authStoreService, RBACService rbacService) {
+        this.usersToActionsMap = new HashMap<>();
+        this.authStoreService = authStoreService;
+        this.rbacService = rbacService;
+    }
+
+    @Override
+    public synchronized void addUser(String loginId, List<String> actionIds) throws RBACException {
+        usersToActionsMap.put(loginId, actionIds);
+    }
+
+    @Override
+    public synchronized boolean isActionPermitted(String loginId, String actionId) throws RBACException {
```

Comment:
Think we just have to initially not have a map in memory as we will get divergence between the different pods.